### ISSUE TITLE
[Feature] Support for >2 prefixes + mention prefixes

### ISSUE THREAD

**Issue by solonovamax** [jagrosh/JDA-Utilities#124]
[download]: https://bintray.com/jagrosh/maven/JDA-Utilities/_latestVersion
[guild]: https://discord.gg/0hMr4ce0tIk3pSjp
[stack overflow]: https://stackoverflow.com/questions/tagged/java
[issues]: https://github.com/JDA-Applications/JDA-Utilities/issues
[jda-issues]: https://github.com/DV8FromTheWorld/JDA/issues
[examples-module]: https://github.com/JDA-Applications/JDA-Utilities/tree/master/examples
[jda-u-projects]: https://github.com/JDA-Applications/JDA-Utilities#projects

#### Issue Checklist
Please follow the following steps before opening this issue.<br>
Issues that do not complete the checklist may be closed without any help.

- [x] I have checked for similar issues on the [issue tracker][issues].
- [x] I have updated to the [latest version][download] of JDA-Utilities.
- [x] I have checked the branches or the maintainers' PRs for upcoming features/bug fixes.

#### Issue Information
Check all that apply:

- [ ] This is a bug report about an error, issue, or bug in JDA-Utilities.
  - [ ] I have been able to consistently reproduce this bug.
- [x] This is a feature request for the JDA-Utilities library.

#### Description

There's a few things I'd like to do with prefixes that I'm currently not able to do.
Here's a list of things I'd like to be able to do:

- More than 2 prefixes. Pretty simple to implement. Just make `#setAlternativePrefix(String)` take a vararg (or add a new method, `#setAlternativePrefixes(String...)`, and then when parsing for the prefix, iterate over the array. Not super complicated. (eg. I want to set the prefix of my bot to `pb `, `poly `, and `$`.)
- Allow enabling/disabling of the mention prefix.
  - Currently, it is unclear about whether the mention prefix is enabled if you provide a prefix. It says it's enabled if you pass `null` to `#setPrefix(String)`, but not what happens if you don't set it to null.
  - I propose adding a new method to the builder, something along the lines of `#enableFeatures(Feature...)`, where `Feature` is an enum. It would allow more granular control for certain true/false boolean features, as opposed to having to add new methods for each one. An example of such a feature would be `Feature.MENTION_PREFIX`. When enabled, it would allow `@TheBot#1234 command` _or_ `@TheBot#1234command` (note the lack of space in the second one) to be used as the prefix. (I'm using `@TheBot#1234`, but I know that mentions are technically `<@123456789...>`. It just doesn't look as nice)

I thought I had a few more, but brain go brrr and now I can't remember them. I'll update the issue if I think of the other 2.

**Comment by jagrosh** [comment]
You can already have as many prefixes as you want by utilizing a guild settings manager. The methods for setting a prefix/altprefix are simply convenience methods.

**Comment by jagrosh** [comment]
Additionally, you can already use mentions as a prefix; set a prefix to `@mention` (that literal string), which is a special case that looks for mentions.

**Comment by solonovamax** [comment]
> You can already have as many prefixes as you want by utilizing a guild settings manager. The methods for setting a prefix/altprefix are simply convenience methods.

Okay, well, that still doesn't discard my point of wanting an _easy_ way to do it.

> Additionally, you can already use mentions as a prefix; set a prefix to @mention (that literal string), which is a special case that looks for mentions.

And this is documented where ??
No offense, but please, document it somewhere. Also, can you enable the mention prefix _alongside_ a normal prefix?

**Comment by jagrosh** [comment]
> Okay, well, that still doesn't discard my point of wanting an easy way to do it.

Using a guild settings manager to add additional prefixes is not difficult, and is the correct approach to wanting anything remotely niche regarding prefixes. The only reason that "altprefix" exists is to be backwards compatible with before the guild settings manager existed; if guild settings manager had existed earlier, the altprefix would not exist at all, as it's a mediocre solution to multiple prefix support.

> And this is documented where ??

While the functionality of `@mention` is not specifically documented, using a mention as a prefix is clearly documented [here](https://jda-applications.github.io/JDA-Utilities/com/jagrosh/jdautilities/command/CommandClientBuilder.html#setPrefix-java.lang.String-). You can still set an alternative prefix and/or utilize the guild settings manager in addition to the default mention prefix. 

TLDR: Use the default prefix (which is a mention) and use a guild settings manager to manage additional global and/or per-guild prefixes.

**Comment by solonovamax** [comment]
Using a guild settings manager to add default prefixes does not sound like a nice solution at all though.
The use of a guild settings manager should be for _guild settings_.
